### ISSUE TITLE
Have a different header and footer for certain products and categories within the same magento store

### ISSUE THREAD

**Issue by support-smartebusiness** [nexcess/magento-turpentine#1464]
Hi,

I am having a slight issue, as some of our products and categories have a custom page header and footer. Basically we have set a custom theme for these products and categories using the custom design attribute in magento. Since varnish is caching our main header and footer on these pages, we are unable to see the custom header and footer. At the moment I have excluded these pages from varnish to avoid issues. However I would like to get varnish working on these pages to.

In short what I'm really looking for is to have two headers and footers on the site. Any advice on how we can achieve this with the turpentine module will be helpful.

Thanks,
Ash

**Comment by m-overlund** [comment]
You need to use ESI Hole punching, using registry keys for products and categories, if that's where the footer/header differs.

See:
https://github.com/nexcess/magento-turpentine/wiki/ESI_Cache_Policy

**Comment by support-smartebusiness** [comment]
Great, Thanks for the suggestion. I was able to solve the issue using the registry keys as mentioned.
